src/nodes.coffee

```diff
@@ -1739,7 +1739,9 @@ exports.JSXElement = class JSXElement extends Base
               new JSXExpressionContainer emptyExpression, locationData: element.locationData
             else
               unwrapped = expression.unwrapAll()
-              if unwrapped instanceof JSXElement
+              if unwrapped instanceof JSXElement and
+                  # distinguish `<a><b /></a>` from `<a>{<b />}</a>`
+                  unwrapped.locationData.range[0] is element.locationData.range[0]
```

Comment:
Both the "unwrapped" and "wrapped" nested JSX element cases currently generate identical node class structures (they're both treated as `Interpolation`s), but we can distinguish them by inspecting location data